.github/workflows/release.yml

```diff
@@ -18,49 +21,86 @@ jobs:
           token: ${{ secrets.flexgetbot_pat }}
           persist-credentials: true
           fetch-depth: 0
-      - name: Setup Git User
+      - name: Error if running on a branch other than develop
         run: |
-          git config user.email github-actions[bot]@users.noreply.github.com
-          git config user.name github-actions[bot]
+          if [ "$(git rev-parse HEAD)" != "$(git rev-parse origin/develop)" ]; then
+            echo "Release script should be run from develop branch."
+            exit 1
+          fi
+      - name: Cancel workflow run if there are no new commits
```

Comment:
Move the logic that skips deployment earlier to enable fail-fast behavior and reduce the complexity of subsequent conditional logic.